src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    /**
+     * Calculate the cost of a journey. Where free transfers are not permitted the cash price is used. If free transfers
+     * are applicable, the most expensive discount fare across all legs is added to the final cumulative price.
+     *
+     * The computed fare for Orca card users takes into account realtime trip updates where available, so that, for
+     * instance, when a leg on a long itinerary is delayed to begin after the initial two hour window has expired,
+     * the calculated fare for that trip will be two one-way fares instead of one.
+     */
+    @Override
+    public boolean populateFare(Fare fare,
+                                   Currency currency,
+                                   Fare.FareType fareType,
+                                   List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules
+    ) {
+        Long freeTransferStartTime = null;
+        float cost = 0;
+        float orcaFareDiscount = 0;
+        for (Ride ride : rides) {
+            RideType rideType = classify(ride.routeData, ride.trip.getId());
+            boolean ridePermitsFreeTransfers = permitsFreeTransfers(rideType);
+            if (freeTransferStartTime == null && ridePermitsFreeTransfers) {
+                // The start of a free transfer must be with a transit agency that permits it!
+                freeTransferStartTime = ride.startTime;
+            }
+            float singleLegPrice = getRidePrice(ride, fareType, fareRules);
+            float legFare = getLegFare(fareType, rideType, singleLegPrice, ride.routeData);
+            boolean inFreeTransferWindow = inFreeTransferWindow(freeTransferStartTime, ride.startTime);
+            if (hasFreeTransfers(fareType, rideType) && inFreeTransferWindow) {
+                // If using Orca (free transfers), the total fare should be equivalent to the
+                // most expensive leg of the journey.
+                orcaFareDiscount = Float.max(orcaFareDiscount, legFare);
+            } else if (usesOrca(fareType) && !inFreeTransferWindow) {
+                // If using Orca and outside of the free transfer window, add the single leg cash price.
+                cost += singleLegPrice;
+            } else {
+                // If not using Orca, add the agencies default price for this leg.
+                cost += legFare;
+            }
+            if (!inFreeTransferWindow) {
+                // If the trip time has exceeded the free transfer time limit of two hours the rider is required to
+                // purchase a new fare. This also resets the free transfer trip window, applies the orca discount to the
+                // overall cost and then resets the orca fare discount ready for the next transfer window.
+                freeTransferStartTime = ridePermitsFreeTransfers ? ride.startTime : null;
+                cost += orcaFareDiscount;
+                orcaFareDiscount = 0;
+            }
```

Comment:
This bit of code...

```java
            } else if (usesOrca(fareType) && !inFreeTransferWindow) {
                // If using Orca and outside of the free transfer window, add the single leg cash price.
                cost += singleLegPrice;
            }
```

...would not end up resetting the `orceFareDicsount` for use in future legs. This means that if the $10 Kitsap Fast Ferry were used for example, that $10 wouldn't cover further leg fares since the `orceFareDiscount` wouldn't be set to this amount. It seems like maybe the `if (!inFreeTransferWindow) {` clause handling the adding of the `orcaFareDiscount` could be moved into here. See example that I haven't tested, but think would work:

```suggestion
           } else if (usesOrca(fareType) && !inFreeTransferWindow) {
                // If using Orca and outside of the free transfer window, add the cumulative Orca fare (the maximum leg 
                // fare encountered within the free transfer window).
                cost += orcaFareDiscount;
                
                // Reset the free transfer start time and next Orca fare as needed.
                if (ridePermitsFreeTransfers) {
                    // The leg is using a ride type that permits free transfers. 
                    // The next free transfer window begins at the start time of this leg.
                    freeTransferStartTime = ride.startTime;
                    // Reset the Orca fare to be the fare of this leg.
                    orcaFareDiscount = legFare;
                } else {
                    // The leg is not using a ride type that permits free transfers.
                    // Since there are no free transfers for this leg, increase the total cost by the fare for this leg.
                    cost += legFare;
                    // The current free transfer window has expired and won't start again until another leg is 
                    // encountered that does have free transfers.
                    freeTransferStartTime = null;
                    // The previous Orca fare has been applied to the total cost. Also, the non-free transfer cost has
                    // also been applied to the total cost. Therefore, the next Orca cost for the next free-transfer 
                    // window needs to be reset to 0 so that it is not applied after looping through all rides.
                    orcaFareDiscount = 0;
                }
            } else {
                // If not using Orca, add the agencies default price for this leg.
                cost += legFare;
            }
        }
```